jme3-materialeditor/src/com/jme3/gde/materialdefinition/EditableMatDefFile.java

```diff
@@ -1,408 +1,403 @@
-    }
-
-    private void registerListener(Statement sta) {
-        if (sta instanceof UberStatement) {
-            ((UberStatement) sta).addPropertyChangeListener(WeakListeners.propertyChange(changeListener, ((UberStatement) sta)));
-        } else if (sta instanceof LeafStatement) {
-            ((LeafStatement) sta).addPropertyChangeListener(WeakListeners.propertyChange(changeListener, ((LeafStatement) sta)));
-        }
-        if (sta.getContents() != null) {
-            for (Statement statement : sta.getContents()) {
-                registerListener(statement);
-            }
-        }
-    }
-
-    public void buildOverview(ExplorerManager mgr) {
-        if (materialDef != null) {
-            mgr.setRootContext(new MatDefNode(lookup));
-
-        } else {
-            mgr.setRootContext(Node.EMPTY);
-        }
-    }
-
-    public String getShaderCode(String version, Shader.ShaderType type) {
-        try {
-            material.selectTechnique(currentTechnique.getName(), SceneApplication.getApplication().getRenderManager());
-            Shader s;
-            StringBuilder sb = new StringBuilder();
-            TechniqueDef def = material.getActiveTechnique().getDef();
-            sb.append(def.getShaderPrologue());
-            material.getActiveTechnique().getDynamicDefines().generateSource(sb, Arrays.asList(def.getDefineNames()), Arrays.asList(def.getDefineTypes()));
-            
-            if (version.equals(GLSL100)) {
-                glsl100.initialize(material.getActiveTechnique().getDef());
-                s = glsl100.generateShader(sb.toString());
-            } else {
-                glsl150.initialize(material.getActiveTechnique().getDef());
-                s = glsl150.generateShader(sb.toString());
-            }
-            for (Shader.ShaderSource source : s.getSources()) {
-                if (source.getType() == type) {
-                    return source.getSource();
-                }
-            }
-            return "";
-        } catch (Exception e) {
-            Exceptions.printStackTrace(e);
-            return "Error generating shader: " + e.getMessage();
-        }
-    }
-
-    public TechniqueBlock getCurrentTechnique() {
-        return currentTechnique;
-    }
-    
-    public void setCurrentTechnique(TechniqueBlock tech){
-        this.currentTechnique = tech;
-    }
-
-    public MatDefBlock getMatDefStructure() {
-        return matDefStructure;
-    }
-    private final MatStructChangeListener changeListener = new MatStructChangeListener();
-    J3MLoader loader = new J3MLoader();
-
-    private void updateLookupWithMaterialData(MatDefDataObject obj) {
-        obj.getLookupContents().add(materialDef);
-        material = new Material(materialDef);
-
-        try {
-            //material.selectTechnique("Default", SceneApplication.getApplication().getRenderManager());
-            if (matToRemove != null) {
-                for (MatParam matParam : matToRemove.getParams()) {

```

Comment:
These lines prevent shader nodes from being updated, since every time you update the MatDet, the values are overwritten by the Material. We're not editing materials, so should be safe to remove?